README.md

```diff
@@ -0,0 +1,123 @@
+
+## Technologies Used
+
+This project leverages a variety of modern technologies and tools to ensure a seamless and efficient experience:
+
+- **Spring Boot**: For creating a stand-alone, production-grade Spring-based application.
+- **Spring Security**: To handle authentication and authorization.
+- **Spring Data JPA**: For simplified data access and management.
+- **Swagger**: For API documentation and testing.
+- **Maven**: For project build and dependency management.
+- **Docker**: To containerize the application, ensuring consistency across different environments and simplifying deployment.
+
+## Project Structure and Functionality
+
+The project is organized into several key components, each responsible for specific functionalities:
+
+### Controllers
+
+- **AuthController**: Manages user-related operations such as registration, login, and profile management.
+- **BookController**: Manages book processing, including book creation and updating for admins, and viewing for users.
+- **OrderController**: Manages order processing, including order creation, updating, and viewing order history.
+- **CategoryController**: Manages category processing, including category creation, updating, and viewing.
+- **ShoppingCartController**: Manages shopping cart processing, including creation, updating, and viewing items (books and other products).
+
+### Services
+
+- **OrderService**: Manages the order processing logic, including placing orders in the database, retrieving order history, and calculating the total price of all items.
+- **BookService**: Manages the book processing logic, including book management for admins and book retrieval for users.
+- **CategoryService**: Manages category processing logic, including category management for admins and category retrieval for users.
+- **ShoppingCartService**: Handles the business logic related to shopping carts, including adding items and updating the cart.
+- **UserService**: Contains business logic for user operations such as registration and user retrieval by username (email).
+
+### Repositories
+
+- **BookRepository**: Interface for CRUD operations on Book entities and special methods for finding books by category.
+- **ShoppingCartRepository**: Interface for CRUD operations on ShoppingCart entities and methods for identifying a shopping cart by user ID.
+- **CartItemRepository**: Interface for CRUD operations on CartItem entities.
+- **OrderRepository**: Interface for CRUD operations on Order entities.
+- **OrderItemRepository**: Interface for CRUD operations on OrderItem entities and methods for retrieving order items by order ID.
+- **UserRepository**: Interface for CRUD operations on User entities and methods for checking if the user is already registered.
+- **RoleRepository**: Interface for CRUD operations on Role entities, including methods for identifying user roles.
+
+## Setup and Usage
```

Comment:
let's mention that we have 2 ways:
* docker
* usual spring boot run

We should also mention:
* If we want to used the usual SB run - we need to have several tools installed, like: database, maven, java etc. and corresponding changes being made to the application.properties - like db URL etc.
* if we use docker compose - add .env file with necessary variables set(add examples)